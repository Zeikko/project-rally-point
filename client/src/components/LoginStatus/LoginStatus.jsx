import PropTypes from 'prop-types'
import React, { Component } from 'react'
import styled from 'styled-components'
import LoginButton from '../LoginButton/LoginButton'
import LogoutButton from '../LogoutButton/LogoutButton'
import { getUserAction } from '../../actions/user-actions'
import * as propTypes from '../../constants/prop-types'

class LoginStatus extends Component {
  componentDidMount() {
    const { dispatch } = this.props
    dispatch(getUserAction())
  }

  render() {
    const { userState, dispatch } = this.props
    if (userState.isLoading) {
      return null
    }
    if (userState.user) {
      return (
        <Wrapper>
          <Avatar alt={userState.user.displayName} src={userState.user.smallAvatarUrl} />
          <DisplayName>{userState.user.displayName}</DisplayName>
          <LogoutButton dispatch={dispatch} />
        </Wrapper>
      )
    }
    return <LoginButton dispatch={dispatch} />
  }
}

LoginStatus.propTypes = {
  dispatch: PropTypes.func.isRequired,
  userState: propTypes.userState.isRequired,
}

export default LoginStatus

const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
`

const Avatar = styled.img`
  height: 32px;
`

const DisplayName = styled.div`
  line-height: 32px;
  padding: 0 10px;
`
