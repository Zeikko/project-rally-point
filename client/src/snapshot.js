import renderer from 'react-test-renderer'
import 'jest-styled-components'

export function matchComponentToSnapshot(component) {
  const json = renderer.create(component).toJSON()
  expect(json).toMatchSnapshot()
}