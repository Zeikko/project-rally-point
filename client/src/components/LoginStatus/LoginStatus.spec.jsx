import React from 'react'
import { matchComponentToSnapshot } from '../../test/snapshot'
import LoginStatus from './LoginStatus'
import initialState from '../../initial-state'
import normalUser from '../../../../fixtures/normal-user.json'

describe('LoginStatus', () => {
  it('renders initial state correctly', () => {
    matchComponentToSnapshot(<LoginStatus
      dispatch={() => {}}
      userState={initialState.userState}
    />)
  })

  it('renders loading state correctly', () => {
    matchComponentToSnapshot(<LoginStatus
      dispatch={() => {}}
      userState={{ isLoading: true }}
    />)
  })

  it('renders logged in state correctly', () => {
    matchComponentToSnapshot(<LoginStatus
      dispatch={() => {}}
      userState={{
        isLoading: false,
        user: normalUser,
      }}
    />)
  })
})
