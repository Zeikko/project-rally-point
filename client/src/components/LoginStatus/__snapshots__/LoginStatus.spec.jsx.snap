// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginStatus renders initial state correctly 1`] = `
<a
  href="/api/auth/login"
>
  <img
    alt="Login"
    src="/img/steam_login.png"
  />
</a>
`;

exports[`LoginStatus renders loading state correctly 1`] = `null`;

exports[`LoginStatus renders logged in state correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  height: 32px;
}

.c2 {
  line-height: 32px;
  padding: 0 10px;
}

<div
  className="c0"
>
  <img
    alt="Test user"
    className="c1"
    src="http://avatar.test/small"
  />
  <div
    className="c2"
  >
    Test user
  </div>
  <button
    href="/api/auth/login"
    onClick={[Function]}
  >
    Logout
  </button>
</div>
`;
